src/browser/url.rs

```diff
@@ -60,6 +60,18 @@ impl Url {
     }
 }
 
+impl std::fmt::Display for Url {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+        let path = self.path.join("/");
+        match (&self.search, &self.hash) {
+            (Some(search), Some(hash)) => write!(fmt, "{}?{}#{}", path, search, hash),
+            (Some(search), None) => write!(fmt, "{}?{}", path, search),
+            (None, Some(hash)) => write!(fmt, "{}#{}", path, hash),
+            _ => write!(fmt, "{}", path),
+        }
```

Comment:
Should we then instead of custom made Url object use something like https://github.com/servo/rust-url?

Reply:
Ah, right. Found your issue there about wasm size. I think that if we can omit websys call then we should. Why do you think it's error prone? Not that I'm not having the same gut feeling, but still what could go wrong? ;)

Reply:
> Why do you think it's error prone?

Once we add another URL part I can guarantee that we will forget to add it into some `write!` function. And once we want to refactor it - e.g. encode hash - again the same problem. And once we add some order-sensitive part (if it's possible) => `match (&self.search, &self.hash, &self.foo)`.. it will be pain.

> I think that if we can omit websys call then we should.

It sounds like premature optimizations. It does make some sense in patching algorithm; it isn't very important here.

Reply:
I remade it using `websys::Url` and it still has all issues you mentioned, with added `expect` call.
If you have better ideas on how it should be done, please share them.

Reply:
I've had a look at the latest commit code which uses web_sys::Url.

TatriX, when you say it has the same issues do you mean if another URL part is added? 

I think that would affect any solution however I think it is more maintainable and readable with the current commit code below, so i'm happy with this `web_sys::Url` approach:

```rust
        let url = web_sys::Url::new_with_base(&self.path.join("/"), dummy_base_url)
             .expect("cannot create url");
         if let Some(search) = &self.search {
             url.set_search(search);
         }
         if let Some(hash) = &self.hash {
             url.set_hash(hash);
         }
```